docs/Why-Mock.md

```diff
@@ -0,0 +1,138 @@
+# Why Mock?
+
+This document tries to answer two questions:
+
+1. Why you should use Mock instead of lower-level tooling like `rpmbuild`
+2. Why you should not try to reimplement Mock and rather join the project
+
+
+## Mock builds packages in a dedicated environment
+
+Builds are done in a dedicated buildroot that contains only a minimal set of
+packages. As a consequence, build failures are deterministic and reproducible
+because they cannot be affected by something that user installed/configured on
+their host system.
+
+Additionally, building RPM packages is inherently dangerous because specfiles
+are written in a Turing-complete language. Building an untrusted package can
+easily lead to a compromised system. This is also true for self-written
+packages. One small bug like `rpm -rf %{?nonexisting}/` can wipe your entire
+system if you are unwise enough to run `rpmbuild` as `root`. Or at least your
+user home directory.
+
+Mock builds packages inside of `systemd-nspawn` containers to prevent such
+disasters from happening.
+
+For truely isolated, offline builds, Mock also supports
+[hermetic builds][hermetic-builds].
```

Comment:
While hermetic allows 100% offline builds, it doesn't isolate from the host.  The hermetic mode page documents this well, but maybe we can elaborate a bit even here...

Basically, anyone who wants to be 100% safe and isolated should go with `podman run --rm -ti --privileged` wrapper.

Reply:
We should not call it hermetic. It's just a reproducible build mode, not a hermetic build mode.

Reply:
Two most difficult problems in SW industry: Naming & Caching. The chosen "feature name" has been "hermetic" -- mostly to align with the SLSA definition https://slsa.dev/spec/v0.1/requirements#hermetic ... I am not against changing the feature name long term, but that would be a different PR than this one.